File: inngest-spring-boot-demo/src/main/java/com/inngest/springbootdemo/FollowupFunction.java
package com.inngest.springbootdemo;

import com.inngest.*;
import org.jetbrains.annotations.NotNull;

import java.util.Map;

public class FollowupFunction extends InngestFunction {

    @NotNull
    @Override
    public InngestFunctionConfigBuilder config(InngestFunctionConfigBuilder builder) {
        return builder
            .id("fn-follow-up")
            .name("My follow up function!")
            .triggerEvent("user.signup.completed")
            .triggerEvent("random-event");
    }

    @Override
    public Map<String, Object> execute(@NotNull FunctionContext ctx, @NotNull Step step) {
        System.out.println("-> follow up handler called " + ctx.getEvent().getName());
        return ctx.getEvent().getData();
    }
}

File: inngest-test-server/src/main/kotlin/com/inngest/testserver/FollowupFunction.kt
package com.inngest.testserver

import com.inngest.*

@FunctionConfig(id = "fn-follow-up", name = "My follow up function!")
@FunctionEventTrigger(event = "user.signup.completed")
@FunctionEventTrigger(event = "random-event")
class FollowupFunction : InngestFunction() {
    override fun execute(
        ctx: FunctionContext,
        step: Step,
    ): LinkedHashMap<String, Any> {
        println("-> follow up handler called " + ctx.event.name)
        return ctx.event.data
    }
}
